src/sys/socket/mod.rs

```diff
@@ -1901,15 +1901,16 @@ unsafe fn read_mhdr<'a, 'i, S>(
         }
     };
 
-    // Ignore errors if this socket address has statically-known length
-    //
-    // This is to ensure that unix socket addresses have their length set appropriately.
-    let _ = unsafe { address.set_length(mhdr.msg_namelen as usize) };
+    let addr_len = mhdr.msg_namelen;
+
+    let address = unsafe {
+        S::from_raw(address.as_ptr().cast(), Some(addr_len))
```

Comment:
Are you aware that this creates a data copy?  The original code was carefully designed to avoid such a thing.

Reply:
Why does it matter that some other function also performs a data copy?  Are you pointing that out to suggest that we improve it?  I'm requesting that you not add an additional data copy to `recvmsg`.

Reply:
Ah okay yeah, that makes sense, I think I misunderstood you before. I'll change the signature of `read_mhdr` back to expecting a pointer, and perform the conversion there.